packages/samples/headless-react/src/components/triggers/redirection-trigger.fn.tsx

```diff
@@ -0,0 +1,26 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {RedirectionTrigger as HeadlessRedirectionTrigger} from '@coveo/headless';
+
+interface HeadlessRedirectionTriggerProps {
+  controller: HeadlessRedirectionTrigger;
+}
+
+export const RedirectionTrigger: FunctionComponent<HeadlessRedirectionTriggerProps> = (
+  props
+) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  if (!state) {
+    return null;
+  }
+
+  if (state.redirectTo) {
+    window.location.href = state.redirectTo;
+    return null;
+  }
+
+  return <div></div>;
```

Comment:
I think can replace these lines with a simple `return null`. No need for a state hook too,

```
const redirect = () => {
     window.location.href = controller.state.redirectTo;
}

useEffect(() => controller.subscribe(redirect), []);

return null;
```